src/main/scala/com/evolutiongaming/cluster/sharding/ShardingStrategy.scala

```diff
@@ -70,6 +70,61 @@ object ShardingStrategy {
   }
 
 
+  /**
+    * Per-shard allocation and rebalance based on matching shard's "role" with node roles
+    */
+  class FilterByRole(
+    shardRole: Shard => Option[String],
```

Comment:
I think we need to change this to `shardRole: (Shard, Address or Requester ) => Option[Role]`

Reply:
Why do we need `Address or Requester` to determine the shard role?

Reply:
This would be required for one of solutions for the corner case when we can't determine the role based on shard (id) as entry does not exist yet. And we could use ActorPath for this.

Reply:
I think that in this case (if `Role` exists in `ActorPath`) we can first add this `ShardId -> Role` mapping to the cache and then proceed with the same flow as in general.

Reply:
add to function passed via constructor? yes this is possible, but don't think this is optimal